packages/core/content-type-builder/admin/src/components/FormModalHeader/index.js

```diff
@@ -110,7 +111,7 @@ const FormModalHeader = ({
   return (
     <ModalHeader>
       <Stack horizontal spacing={3}>
-        <AttributeIcon type={icon} />
+        <AttributeIcon type={icon} customField={customFieldUid} />
```

Comment:
Mh, I'm not sure I like that. Why? Because you already introduce some custom fields conditionals here, so why not picking the icon here from `customFieldsRegistry` and then pass it down as `type={}`?

Reply:
I think `type` expects a string that it uses as a lookup in `AttributeIcon`. Since an icon isn't required on a custom field we want to use that type's icon if the user did not register their custom field with an icon. That logic is handled in the `AttributeIcon` component